sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+
+* _Development Pipeline Impact. It is assumed that a sub-project may already be
+  subject to the parent project’s decisions so link to, or briefly describe the
+  testing and review processes that the software undergoes as it is developed
+  and built. Of interest might be specific information as to whether
+  contributors are required to sign commits, if project artifacts/images are
+  immutable and signed, any automated checks for vulnerabilities, etc._
+* _Communication Channels. Reference where you document how to reach your main
+  project team and sub-project leads, for example:_
+  * _Internal. How do team members communicate with each other?_
+  * _Inbound. How do users or prospective users communicate with the team?_
+  * _Outbound. How do you communicate with your users? (e.g. flibble-announce@
+    mailing list)_
+* _Ecosystem. How does your software fit into the main project and other cloud
+  native ecosystems?_
+
+#### Development Workflow
+
+Cluster API’s development workflow is documented within the repository’s
+[CONTRIBUTING.md](https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md)
+
+Key processes include:
+
+* All source code is publicly available in Github
+* All pull requests trigger jobs that perform:
+  * Linting, including gosec static code analysis
+  * Unit tests
+  * E2E tests using nested Docker as a simulated infrastructure provider
+  * Checks for breaking API changes
+  * Tests for broken URLs in generated documentation
+  * PRs cannot be merged without passing tests without an explicit /override
+    from a maintainer
+* Each PR requires approval from maintainers listed in the OWNERS file
+* For all Cluster API subprojects, contributors are required to have signed the
+  CNCF CLA, but no cryptographic signatures are required for individual commits.
+* Pushes to the main branch are forbidden by convention
+* Release process is almost fully automated via Github actions
+* Checksum, signature and SPDX files are not currently generated for any
+  artifacts
+* Container images are produced automatically by Google CloudBuild, and Cluster
+  API uses the upstream Kubernetes image promotion process to promote individual
+  OCI SHAs from staging to release.
+* Deployment manifests reference image tags rather than SHA digests
+* ACLs for ownership of the repository and approvals are configured in code
+  within the OWNERS file at the top of the repository as well as configured via
+  the github.com/kubernetes/org repository.
+* For Cluster API Provider AWS since v0.6.7 and Cluster API Provider Azure since
+  v0.4.13 maintainers do GPG sign all tags, and maintainers are expected to
+  maintain their GPG keys with public keys verifiable in Github. Cluster API
+  Core repository has tags signed by Github Actions since v1.0.3
+
+#### Identified gaps in development workflow
+
+##### [DEV-01: Outdated OWNERS file](https://github.com/kubernetes-sigs/cluster-api/issues/6151)
```

Comment:
I'm aware of work in main Kubernetes to ensure the proper hygiene of owner files. As far as you know, Is it possible to leverage on the same tooling?

Reply:
Resolving this thread, as the discussions have begun at K8s project level for now and will continue them there